### PR TITLE
[FEATURE] Add 'sap-csp-policies' option to 'serve' command.

### PR DESCRIPTION
**Pull request by codeworrior**
**Thank you for your contribution!** 🙌

To get it merged faster, kindly review the checklist below:

## Pull Request Checklist
- [ ] Reviewed the [Contributing Guidelines](https://github.com/SAP/ui5-tooling/blob/master/CONTRIBUTING.md#-contributing-code)
    + Especially the [How to Contribute](https://github.com/SAP/ui5-tooling/blob/master/CONTRIBUTING.md#how-to-contribute) section 
- [ ] [No merge commits](https://github.com/SAP/ui5-tooling/blob/master/docs/Guidelines.md#no-merge-commits)
- [ ] [Correct commit message style](https://github.com/SAP/ui5-tooling/blob/master/docs/Guidelines.md#commit-message-style)

### CODE REVIEW

**Comment by codeworrior** [comment]
Note: this PR is a counterpart to https://github.com/SAP/ui5-server/pull/179.

**Comment by codeworrior** [comment]
There's an open discussion about the name of the new option (`--csp-defaults`). At least in the tests, this looks a bit confusing. Other ideas had been `--csp-presets`, or `--csp-sap-presets` etc.

Until we come to a conclusion, I won't submit neither this PR nor SAP/ui5-server#179.

**Comment by RandomByte** [comment]
Why not simply `--csp`?

**Comment by codeworrior** [comment]
To me, `--csp` sounds as if this option would be mandatory to enable CSP at all. But that's not true.  Even without any option, selection and definition of policies is already possible (via URL parameter `--sap-ui-xx-csp-policy`).

The purpose of the new option is to enable a set of policies that SAP want's to use in-house (for UI5 DIST layer development). As we don't have means to activate configuration for such a limited scope, having a new option seemed to be the 2nd best approach to me. But the term 'defaults' somehow conflicts with the fact that there are already (other) defaults even without the new option.

**Comment by RandomByte** [comment]
Since CLI flags have defaults themselves, I would refrain from including the string "default" in any flags name since it refers to another level of defaults. Which I find very confusing.

From a user perspective the URL parameters are mostly unknown. Even when known, I would see them as a separate feature. I.e. a standard functionality of the server.

This new feature of having a certain set of CSP headers being sent from the server for **every** request can *imho* be referred to by a simple "do CSP" toggle. However, since it seems to be an in-house feature, maybe `--sap-csp` or `--sap-csp-policies` (to relate the flag name from the URL parameter name).

I don't see a reason for separating this feature from the existing URL parameters from a wording perspective. They are both controlled in different ways anyways. 

Maybe read it like this:
> When setting the flag `--sap-csp-policies`, the server always sends CSP headers `sap-target-level-1` and `sap-target-level-2` in report only mode

**Comment by codeworrior** [comment]
> Since CLI flags have defaults themselves, I would refrain from including the string "default" in any flags name since it refers to another level of defaults. Which I find very confusing.

That was exactly my point why I wasn't happy with the name of the option. To me it became obvious in the test code where I was tempted to name two tests "CSP defaults", one for each level of defaults.

I doubt that '_from a user perspective, the URL parameters are mostly unknown_'. At least internally,  developers have been asked to use these parameters to test their code against CSPs. Our QUnit integration even supports a UI (drop down) for them.

But it doesn't really matter as I'm totally fine with `--sap-csp-policies`. I'll even use 
>always send CSP headers sap-target-level-1 and sap-target-level-2 in report only mode

for the CLI help.

File: lib/cli/commands/serve.js [hunk 1]
```diff
@@ -37,6 +37,11 @@ serve.builder = function(cli) {
 			default: "$HOME/.ui5/server/server.crt",
 			type: "string"
 		})
+		.option("sap-csp-policies", {
+			describe: "always send content security policies 'sap-target-level-1' and 'sap-target-level-2' in report-only mode",
```
Comment (RandomByte) [inline lib/cli/commands/serve.js]: Capital A at the start of the sentence

File: lib/cli/commands/serve.js [hunk 2]
```diff
@@ -37,6 +37,11 @@ serve.builder = function(cli) {
 			default: "$HOME/.ui5/server/server.crt",
 			type: "string"
 		})
+		.option("sap-csp-policies", {
```
Comment (RandomByte) [inline lib/cli/commands/serve.js]: The README.md needs to be updated with the new `--help` output for this flag as well.

Comment (RandomByte) [reply lib/cli/commands/serve.js]: README.md still needs to be updated

Comment (RandomByte) [reply lib/cli/commands/serve.js]: Done with 8d5f8fd2f2a95a56372831aed1112d012b9c4878

**Comment by RandomByte** [comment]
@svbender for final approval.

**Please use Squash and merge since the commit messages do not align with our convention. The squash commit should have a `[FEATURE]` prefix**

**Pull request merged**